funcx_endpoint/funcx_endpoint/endpoint/rabbit_mq/result_queue_publisher.py

```diff
@@ -0,0 +1,86 @@
+import logging
+
+import pika
+
+logger = logging.getLogger(__name__)
+
+
+class ResultQueuePublisher:
+    """ResultPublisher publishes results to a topic EXCHANGE_NAME, with
+    the {endpoint_id}.results as a routing key.
+    """
+
+    EXCHANGE_NAME = "results"
+    EXCHANGE_TYPE = "topic"
+    QUEUE_NAME = "results"
+    GLOBAL_ROUTING_KEY = "*.results"
+
+    def __init__(
+        self,
+        endpoint_id: str,
+        pika_conn_params: pika.connection.Parameters,
+    ):
+        """
+        Parameters
+        ----------
+        endpoint_uuid: str
+            Endpoint UUID string used to identify the endpoint
+        pika_conn_params: pika.connection.Parameters
+            Pika connection parameters to connect to RabbitMQ
+        """
+        self.endpoint_id = endpoint_id
+        self.params = pika_conn_params
+        if self.params._heartbeat is None:
+            self.params._heartbeat = 0
+
+        self._channel = None
+        self._connection = None
+
+        self.routing_key = f"{self.endpoint_id}.results"
+
+    def connect(self) -> pika.BlockingConnection:
+        """Connect
+
+        :rtype: pika.BlockingConnection
+        """
+        logger.info(f"Connecting to {self.params}")
+        self._connection = pika.BlockingConnection(self.params)
+        self._channel = self._connection.channel()
+        self._channel.confirm_delivery()
+        self._channel.exchange_declare(
+            exchange=self.EXCHANGE_NAME, exchange_type=self.EXCHANGE_TYPE
+        )
+
+        self._channel.queue_declare(queue=self.QUEUE_NAME)
```

Comment:
It appears that use of `passive=True` was reverted in https://github.com/funcx-faas/funcX/pull/701/commits/43ef7a683a9860c0b52827192e0f732d39dc0913 .

I think we need to discuss this. The whole point of setting `passive=True` is that the endpoint should _not_ be trying to create the `"results"` queue. It should only try to ensure that queue exists and then use it.
(As opposed to the queues named by endpoint_id.)

Reply:
Sorry, this got mixed up. This is now fixed, with a fixture added to tests that ensure that a results queue exists with the correct routing key binding.